### PR TITLE
FPLA-116: Add-Documents

### PR DESCRIPTION
**Pull request by Jamesnelson117**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-116

### Change description ###

Added three new documents to the 'Documents' event - 

1. Social work evidence template (SWET)
2. Threshold document
3. Checklist document

Document statuses now extended to include 'Included in social work evidence template (SWET)' (Not in 1). Social work evidence template (SWET))

PDF updated to support new document types and to support existing additional documents.
Validation added to ensure additional documents added provide a name.
e2e / int tests updated to support new documents

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+package uk.gov.hmcts.reform.fpl.controllers;
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: could you write a test to see what happens when you have multiple documents uploaded with no titles

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(callbackResponse.getErrors()).contains("You must give additional document 1 a name.");
+    }
+
+    @Test
+    void shouldReturnWithErrorsIfMultipleDocumentTitlesHaveNotBeenProvided() throws Exception {
+        CallbackRequest request = CallbackRequest.builder().caseDetails(CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
+                .put("documents_socialWorkOther", ImmutableList.builder()
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: ImmutableMap.of() and ImmutableList.of() will solve your endless lines of .build here!

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = MAPPER.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(callbackResponse.getErrors()).isEmpty();
+    }
+
+    private CallbackRequest createCallbackRequest(String documentTitle) {
+        return CallbackRequest.builder().caseDetails(CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: ImmutableMap.of() and ImmutableList.of()!

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherSubmissionController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        List<String> validationErrors = new ArrayList<String>();
+
+        if (!socialWorkOtherData.isEmpty()) {
+            AtomicInteger i = new AtomicInteger(1);
+
+            socialWorkOtherData.forEach(document -> {
+                Map<String, Object> documentValue = (Map<String, Object>) document.get("value");
+                Optional documentTitle = Optional.ofNullable(documentValue.get("documentTitle"));
+
+                if (documentTitle.isEmpty() || "".equals(documentValue.get("documentTitle"))) {
```
Comment (swalker125) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherSubmissionController.java]: .isBlank() would help here I think. thoughts?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(callbackResponse.getErrors()).contains("You must give additional document 1 a name.");
+    }
+
+    @Test
+    void shouldReturnWithErrorsIfMultipleDocumentTitlesHaveNotBeenProvided() throws Exception {
+
+        CallbackRequest request = CallbackRequest.builder().caseDetails(CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: can this ImmutableMap.builder() be replaced with .of as well?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                        )
+                    ),
+                    ImmutableMap.of(
+                        "id", "12345",
+                        "value", ImmutableMap.of(
+                            "documentTitle", ""
+                        )
+                    )
+                )).build())
+            .build()).build();
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: let's space down each .build() as well

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+        MvcResult response = performResponseCallBack(request);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = MAPPER.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(callbackResponse.getErrors())
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: .contains() accepts multiple I believe, so we could assert both in one line. Can we also do something like containsExactly? or are other errors present in that response?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = MAPPER.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(callbackResponse.getErrors()).isEmpty();
+    }
+
+    private CallbackRequest createCallbackRequest(String documentTitle) {
+        return CallbackRequest.builder().caseDetails(CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: ImmutableMap.builder -> of here also?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+                .put("documents_socialWorkOther", ImmutableList.of(
+                    ImmutableMap.of(
+                        "id", "12345",
+                        "value", ImmutableMap.of(
+                            "documentTitle", documentTitle
+                        )
+                    )
+                )).build())
+            .build()).build();
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: let's space down the second .build() here also

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java (hunk 8 above)
Comment (Jamesnelson117) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: Good spot

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+    @Test
+    void shouldReturnWithErrorsIfDocumentTitleIsNotPopulated() throws Exception {
+        CallbackRequest request = createCallbackRequest("");
+
+        MvcResult response = performResponseCallBack(request);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = MAPPER.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(callbackResponse.getErrors()).contains("You must give additional document 1 a name.");
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherControllerTest.java]: let's go containsExactly here also

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherSubmissionController.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    @SuppressWarnings("unchecked")
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        List<Map<String, Object>> socialWorkOtherData = (List<Map<String, Object>>) caseDetails
+            .getData().get("documents_socialWorkOther");
+        List<String> validationErrors = new ArrayList<String>();
+
+        if (!socialWorkOtherData.isEmpty()) {
+            AtomicInteger i = new AtomicInteger(1);
```
Comment (geek-soft) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/SocialWorkOtherSubmissionController.java]: Wired choice...

**Pull request merged**